libs/components/src/next/form/datepicker/vl-datepicker.component.ts

```diff
@@ -0,0 +1,370 @@
+            defaultMinute = Number(this.maxTime.split(':')[1]);
+        }
+
+        const defaultDate = this.parseSelectedDate(this.selectedDate);
+        return {
+            allowInput: !(this.disabled || this.readonly),
+            dateFormat: this.format,
+            defaultHour: defaultHour,
+            defaultMinute: defaultMinute,
+            maxDate: this.maxDate,
+            minDate: this.minDate,
+            defaultDate: defaultDate,
+            enableTime: this.type === 'time' || this.type === 'date-time',
+            noCalendar: this.type === 'time',
+            time_24hr: !this.amPm,
+            minTime: this.minTime,
+            maxTime: this.maxTime,
+            mode: this.type !== 'range' ? 'single' : 'range',
+        };
+    }
+
+    getOptions(): Options {
+        const datepickerButton = this.shadowRoot?.querySelector('button');
+        const datepicker = this.shadowRoot?.querySelector('#wrapper') as HTMLInputElement;
+
+        const staticOptions = {
+            autoFillDefaultTime: true,
+            locale: Dutch.nl,
+            clickOpens: false,
+            onChange: this.handleDateChange,
+            // onValueUpdate: this.handleValueUpdated,
+            positionElement: datepickerButton,
+            static: true,
+            appendTo: datepicker,
+        };
+
+        const options = {
+            ...staticOptions,
+            ...this.getDynamicOptions(),
+        } as unknown as Options;
+
+        Object.keys(options).forEach((key) => {
+            if (options[key as keyof Options] === undefined) delete options[key as keyof Options];
+        });
+        return options;
+    }
+
+    private getDatePicker() {
+        return this.shadowRoot?.querySelector('#wrapper') as HTMLDivElement;
+    }
+
+    private getFlatpickrWrapper() {
+        return this.shadowRoot?.querySelector('.flatpickr-wrapper') as HTMLDivElement;
+    }
+
+    private getFlatpickrCalendar() {
+        return this.shadowRoot?.querySelector('.flatpickr-calendar') as HTMLDivElement;
+    }
+
+    private getVisibleInputElement(): HTMLInputElement | undefined | null {
+        return this.renderRoot?.querySelector('input:not([type="hidden"])') as HTMLInputElement;
+    }
+
+    private updateOptionsForInstance(options: Options) {
+        Object.keys(options)
+            .map((key) => key as keyof Options)
+            .forEach((key) => {
+                this.instance?.set(key, options[key as keyof Options]);
+            });
+    }
+
+    private initializeComponent() {
+        if (this.getDatePicker() && !this.instance) {
+            this.instance = flatpickr(this.getDatePicker(), this.getOptions()) as unknown as Instance;
+            this.getDatePicker().classList.add('static');
+        }
+    }
+
+    private isInShadowDOM() {
+        return this.getRootNode() instanceof ShadowRoot;
+    }
+
+    onInput = (event: Event & { target: HTMLInputElement }) => {
```

Comment:
Kan je een vl-input event dispatchen conform de andere form componenten?

Reply:
ik heb het geplaatst bij `updateValue()` zodat event ook wordt gethrowd bij het resetten, in het geval er een initial value wordt gezet & als de gebruiker in de `calendar`-popup een waarde kiest (gezien `onInput` enkel bij manuele ingave triggert)